posts/Swift-Optionals-Are-Monads-Too.md

```diff
@@ -0,0 +1,264 @@
+Monads, Functors, and Applicative Functors oh my! Have you ever wondered what any of these concepts are and why they are useful? Let's dive into some explanations and then we can see how useful they really can be in your daily programming.
+  
+# Swift's Optional Type
+
+So how does Swift define Optional? Opening up the generated Swift header we see the following:
+
+```swift
+/// A type that can represent either a `Wrapped` value or `nil`, the absence
+/// of a value.
+public enum Optional<Wrapped> : _Reflectable, NilLiteralConvertible {
+    case None
+    case Some(Wrapped)
+    /// Construct a `nil` instance.
+    public init()
+    /// Construct a non-`nil` instance that stores `some`.
+    public init(_ some: Wrapped)
+    /// If `self == nil`, returns `nil`.  Otherwise, returns `f(self!)`.
+    @warn_unused_result
+    public func map<U>(@noescape f: (Wrapped) throws -> U) rethrows -> U?
+    /// Returns `nil` if `self` is `nil`, `f(self!)` otherwise.
+    @warn_unused_result
+    public func flatMap<U>(@noescape f: (Wrapped) throws -> U?) rethrows -> U?
+    /// Create an instance initialized with `nil`.
+    public init(nilLiteral: ())
+}
+```
+
+Reading the first comment in the source code gives us an awesome definition of what an Optional really is. 
```

Comment:
I'm just that excited about the definition. I could run with `great`, `perfect`, or `clear`.